bootstrap/bootstrap-api/src/main/java/com/alipay/sofa/rpc/bootstrap/DefaultConsumerBootstrap.java

```diff
@@ -262,14 +260,12 @@ public List<ProviderGroup> subscribe() {
         String directUrl = consumerConfig.getDirectUrl();
         if (StringUtils.isNotEmpty(directUrl)) {
             // 如果走直连
-            result = subscribeFromDirectUrl(directUrl);
-        } else {
-            // 没有配置url直连
-            List<RegistryConfig> registryConfigs = consumerConfig.getRegistry();
-            if (CommonUtils.isNotEmpty(registryConfigs)) {
-                // 从多个注册中心订阅服务列表
-                result = subscribeFromRegistries();
-            }
+            consumerConfig.setRegistry(new RegistryConfig().setProtocol(REGISTRY_PROTOCOL_DOMAIN));
+        }
+        List<RegistryConfig> registryConfigs = consumerConfig.getRegistry();
+        if (CommonUtils.isNotEmpty(registryConfigs)) {
+            // 从多个注册中心订阅服务列表
+            result = subscribeFromRegistries();
```

Comment:
@OrezzerO There is a little doubt here. When directurl is set, it will also subscribe to addresses from other registries. Will it affect the routing results.

Reply:
Yes ,you're right. Now, when directUrl is set, DomainRegistry is reserved only.